src/misc/debug.h

```diff
@@ -39,13 +43,80 @@ extern void debug_vprintf_trace(const char* func_name,
 				unsigned int line,
 				int level, const char* fmt, va_list ap);
 
+#ifdef __cplusplus
+#  define EXTERN_FUNC extern "C"
+#else
+#  define EXTERN_FUNC extern
+#endif /* __cplusplus */
+
+#ifdef BART_SUPER_DEBUG
+#  define SUPER_DEBUG_OUT(...) BART_OUT(__VA_ARGS__)
+#else
+#  define SUPER_DEBUG_OUT(...) (void)0
+#endif /* BART_SUPER_DEBUG */
+
+#define BART_OUT(...)							\
+     debug_printf_trace(__FUNCTION__, __FILE__, __LINE__, DP_INFO, __VA_ARGS__)
+#define BART_ERR(...)							\
+     debug_printf_trace(__FUNCTION__, __FILE__, __LINE__, DP_ERROR, __VA_ARGS__)
+#define BART_WARN(...)							\
+     debug_printf_trace(__FUNCTION__, __FILE__, __LINE__, DP_WARN, __VA_ARGS__)
+
+
 extern void debug_backtrace(size_t n);
 
 extern void debug_trace(const char* fmt, ...);
 
 #define TRACE()	debug_trace("%s:%d %s\n", __FILE__, __LINE__, __func__)
 
 
+#ifdef USE_LOG_BACKEND
```

Comment:
this should go in the header of the corresponding c/cc file. (and simplied to one function)

Reply:
I prefer not to add an extra header file for just a single function, so I've left the prototype of `vendor_log`in debug.h